crates/fs/src/fs.rs

```diff
@@ -587,6 +596,14 @@ impl Fs for RealFs {
 
         watcher.add(path).ok(); // Ignore "file doesn't exist error" and rely on parent watcher.
 
+        // Check if path is a symlink and follow the target parent
+        if let Ok(target) = self.read_link(&path).await {
```

Comment:
Can we re-use the read_link call above? It should be fine to clone the path into the watcher callback above.